src/tests/tests/background.js

```diff
@@ -357,4 +359,107 @@ QUnit.test("subdomains on the yellowlist are preserved", (assert) => {
   );
 });
 
+QUnit.test("forgetFirstPartySnitches migration properly handles snitch entries with no MDFP entries", (assert) => {
+  const actionMap = badger.storage.getBadgerStorageObject('action_map'),
+    snitchMap = badger.storage.getBadgerStorageObject('snitch_map');
+
+  let snitchNoMDFP = {
+    ['amazon.com']: ['amazonads.com', 'amazing.com', 'amazonrainforest.com']
+  };
+
+  let actionNoMDFP = {
+    'amazon.com': {
+      heuristicAction: "cookieblock",
+      userAction: "",
+      dnt: false,
+      nextUpdateTime: 0,
+    }
+  };
+
+  snitchMap.updateObject(snitchNoMDFP);
+  actionMap.updateObject(actionNoMDFP);
+  migrations.forgetFirstPartySnitches(badger);
+
+  assert.deepEqual(
+    actionMap.getItem('amazon.com'),
+    actionNoMDFP['amazon.com'],
+    "action map preserved for domain with no MDFP snitch entries"
+  );
+
+  assert.ok(badger.storage.getBadgerStorageObject('snitch_map').getItem('amazon.com'),
+    'forget first party migration does not affect a snitchmap with no misattributed MDFP items'
+  );
+});
+
+QUnit.test("forgetFirstPartySnitches migration properly handles snitch entries with some MDFP entries", (assert) => {
+  const actionMap = badger.storage.getBadgerStorageObject('action_map'),
+    snitchMap = badger.storage.getBadgerStorageObject('snitch_map');
+
+  let snitchSomeMDFP = {
+    ['amazon.com']: ['amazon.ca', 'amazon.co.jp', 'amazing.com']
+  };
+
+  let actionSomeMDFP = {
+    'amazon.com': {
+      heuristicAction: "cookieblock",
+      userAction: "",
+      dnt: false,
+      nextUpdateTime: 0,
+    }
+  };
+
+  snitchMap.updateObject(snitchSomeMDFP);
+  actionMap.updateObject(actionSomeMDFP);
+  migrations.forgetFirstPartySnitches(badger);
+
+  assert.equal(
+    badger.storage.getAction('amazon.com'),
+    constants.ALLOW,
+    "Action downgraded for partial MDFP domain"
+  );
+
+  assert.deepEqual(snitchMap.getItem('amazon.com')[0] === 'amazing.com' && snitchMap.getItem('amazon.com').length === 1,
```

Comment:
What I meant earlier is that we should take advantage of `assert.deepEqual()` and simplify the checks here. Instead of checking the first item of the array and the length of the array, we could just compare all of `snitchMap.getItem('amazon.com')` to what we expect it to be.